src/app/persoonsgegevens/Persoonsgegevens.ts

```diff
@@ -42,22 +40,45 @@ export class PersoonsgegevensMapper {
 
   static fromHaalCentraal(data: any): Persoonsgegevens {
 
+    const geboorteDatumIso = data.geboorte.datum.datum as string;
+    const geboortedatum = PersoonsgegevensMapper.isoDateToDutchFormat(geboorteDatumIso);
+
     return {
       bsn: data.burgerservicenummer,
       naam: data.adressering.aanschrijfwijze.naam,
       voorletters: data.naam.voorletters,
       voornamen: data.naam.voornamen,
-      voorvoegsel: data.naam.voorvoegsel,
+      voorvoegsel: data.naam.voorvoegsel ?? '',
       geslachtsnaam: data.naam.geslachtsnaam,
-      geboortedatum: data.geboorte.datum.datum,
-      nederlandseNationaliteit: data.nationaliteiten[0].code == LANDCODE_NEDERLAND ? 'Ja' : 'Nee',
+      geboortedatum: geboortedatum,
+      nederlandseNationaliteit: PersoonsgegevensMapper.hasNederlandseNationaliteit(data.nationaliteiten),
       geslacht: data.geslacht.code,
-      adresHaalCentraal: [
-        data.adressering.adresregel1 ?? '',
-        data.adressering.adresregel2 ?? '',
-        data.adressering.adresregel3 ?? '',
-      ].join('\n'),
+
+      straat: data.verblijfplaats.verblijfadres.officieleStraatnaam,
+      huisnummer: data.verblijfplaats.verblijfadres.huisnummer,
+      postcode: data.verblijfplaats.verblijfadres.postcode,
+      woonplaats: data.verblijfplaats.verblijfadres.woonplaats,
     };
   }
 
+  static hasNederlandseNationaliteit(nationaliteiten?: any[]) {
+    if (!nationaliteiten || nationaliteiten.length == 0) {
+      return 'Nee';
+    }
+    for (const nationaliteit of nationaliteiten) {
+      if (nationaliteit.nationaliteit.code == LANDCODE_NEDERLAND) {
+        return 'Ja';
+      }
+    }
+    return 'Nee';
+  }
+
+  static isoDateToDutchFormat(geboorteDatumIso: string) {
```

Comment:
Ik zou hier normaal aanraden de js-dateformatting-logica te gebruiken. Helaas snapt die een datum als '2024-02-00' (terecht) niet, dus moet je wel bijna. Ik zou wel validatie willen doen op de datum (als in, heeft lengte '0000-00-00'.length, en misschien datum.split('-').length == 3)?

Reply:
We gebruiken nu de langFormat optie om de datum geformateerd weer te geven.